ginga/util/catalog.py

```diff
@@ -229,6 +235,169 @@ def _search(self, center, radius, catalog):
         return results
 
 
+# EBR: add vizier catalog class
+class AstroqueryVizierCatalogServer(AstroqueryCatalogServer):
+    """For queries using the `astroquery.vizier` function."""
+
+    kind = 'astroquery.vizier'
+
+    @classmethod
+    def get_params_metadata(cls):
+        from ginga.misc.ParamSet import Param
+        return [
+            Param(name='ra', type=str, widget='entry',
+                  description="Right ascension component of center"),
+            Param(name='dec', type=str, widget='entry',
+                  description="Declination component of center"),
+            Param(name='width', type=float, default=1, widget='entry',
+                  description="Width of box in degrees"),
+            Param(name='height', type=float, default=1, widget='entry',
+                  description="Height of box in degrees"),
+            Param(name='r', type=float, default=5.0, widget='entry',
+                  description="Radius from center in arcmin"),
+        ]
+
+    # EBR add output colums and filters for vizier catalog
+    def __init__(self, logger, full_name, key, mapping, cat_columns=[], cat_column_filters={}, description=None):
+        super(AstroqueryVizierCatalogServer, self).__init__(logger, full_name,
+                                                            key, None, mapping,
+                                                            description=description, cat_columns=cat_columns, cat_column_filters=cat_column_filters)
+
+    def set_cat_column_filters(self, cat_filter):
+        self.cat_column_filters = cat_filter
+
+    def set_cat_columns(self, cat_columns):
+        self.cat_columns = cat_columns
+
+    def getParams(self):
+        return self.get_params_metadata()
+
+    def _search_radius(self, center, radius, catalog, columns, column_filters):
+        # override this method to pass some special kwargs to the search
+
+        if columns:
+            columns.insert(0, self.mapping['id'])
+            columns.insert(1, self.mapping['ra'])
+            columns.insert(2, self.mapping['dec'])
+
+        v = Vizier(columns=columns, column_filters=column_filters)
+        results = Vizier.query_region(center, radius, catalog=catalog)
+
+        return results[0]
```

Comment:
I needed to change this to:
```python
if len(results) > 0:
    return results[0]
return None
```
Otherwise it would throw an exception if the query returned no results